packages/app-store/_utils/useAddAppMutation.ts

```diff
@@ -113,3 +113,17 @@ function useAddAppMutation(_type: App["type"] | null, allOptions?: UseAddAppMuta
 }
 
 export default useAddAppMutation;
+
+const generateSearchParamString = ({
+  stateStr,
+  teamId,
+  returnTo,
+}: {
+  stateStr: string;
+  teamId?: number;
+  returnTo?: string;
+}) => {
+  const teamIdParam = teamId ? `&teamId=${teamId}` : "";
+  const returnToParam = returnTo ? `&returnTo=${returnTo}` : "";
+  return `?state=${stateStr}${teamIdParam}${returnToParam}`;
+};
```

Comment:
Not a blocker but let's rely on constructing URLs using the the URL class or our own extended one.

https://nodejs.org/api/url.html
https://developer.mozilla.org/en-US/docs/Web/API/URL/URL